server/src/librelay/relay_server.go

```diff
@@ -324,38 +324,35 @@ func (relay *RelayServer) IsUnstaked() (removed bool, err error) {
 	return true, nil
 }
 
-func (relay *RelayServer) RegistrationDate() (when int64, err error) {
+func (relay *RelayServer) BlockCountSinceRegistration() (when uint64, err error) {
 	lastBlockHeader, err := relay.Client.HeaderByNumber(context.Background(), nil)
 	if err != nil {
 		log.Println(err)
 		return
 	}
 	startBlock := uint64(0)
-	if lastBlockHeader.Number.Uint64() > relay.RegistrationBlockRate {
+	lastBlockNumber := lastBlockHeader.Number.Uint64()
+	if lastBlockNumber > relay.RegistrationBlockRate {
 		startBlock = lastBlockHeader.Number.Uint64() - relay.RegistrationBlockRate
 	}
 	filterOpts := &bind.FilterOpts{
 		Start: startBlock,
-		End:   nil,
+		End:   &lastBlockNumber,
 	}
 	iter, err := relay.rhub.FilterRelayAdded(filterOpts, []common.Address{relay.Address()}, nil)
 	if err != nil {
 		log.Println(err)
 		return
 	}
-
+	// We only care about the last registration event
+	for iter.Next() {}
 	if (iter.Event == nil && !iter.Next()) ||
 		(bytes.Compare(iter.Event.Relay.Bytes(), relay.Address().Bytes()) != 0) ||
 		(iter.Event.TransactionFee.Cmp(relay.Fee) != 0) ||
```

Comment:
If you check the txfee when checking for the last event, then it means a relay can't change it's txfee... Remove this check, as it adds nothing (you can argue whether a check for the url is required. I think it isn't, but at least, it's not harmful either, but txfee shouldn't be checked)

Reply:
I actually thought that if you change the tx fee then it makes sense to verify it in the event, why does it mean that you can't change it?